WhateverGreen/kern_igfx.hpp

```diff
@@ -499,6 +499,95 @@ class IGFX {
 		static bool forceWakeWaitAckFallback(uint32_t, uint32_t, uint32_t);
 		static void forceWake(void*, uint8_t set, uint32_t dom, uint32_t);
 	} ForceWakeWorkaround;
+	
+	/**
+	 *  Interface of a submodule to fix Intel graphics drivers
+	 */
+	class PatchSubmodule {
+	public:
+		/**
+		 *  Virtual destructor
+		 */
+		virtual ~PatchSubmodule() = default;
+		
+		/**
+		 *  True if this submodule should be enabled
+		 */
+		bool enabled {false};
+		
+		/**
+		 *  True if this submodule requires patching the framebuffer driver
+		 */
+		bool requiresPatchingFramebuffer {false};
+		
+		/**
+		 *  True if this submodule requires patching the graphics acceleration driver
+		 */
+		bool requiresPatchingGraphics {false};
+		
+		/**
+		 *  Initialize any data structure required by this submodule if necessary
+		 */
+		virtual void init() = 0;
```

Comment:
Do not make (any) functions pure virtual in my opinion. Just provide empty body for them. This will make it much easier to implement in subsequent submodules.

Reply:
Make sense. A concrete type does not have to implement unnecessary functions.
Fixed.